pkg/secretlock/local/passphrase_util.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package local
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/google/tink/go/subtle/random"
+)
+
+// CreateMasterKey will create a new masterkey and returns the encrypted content to be used by Lock
+// The user will then have to store it either in an env variable or a local path prior to using Lock
+// by calling either InitEnv() or InitPath() functions and passing the resulting Reader to New()
+// Calling this method and storing the returned key as mentioned above is a prerequisite for to use
+// the SecretLock service in the Aries agent.
+func CreateMasterKey(passphrase string) (string, error) {
+	masterKey := generateMasterKey(passphrase)
+
+	aeadCipher, err := createAESCipher(masterKey)
+	if err != nil {
+		return "", err
+	}
+
+	nonce := random.GetRandomBytes(uint32(aeadCipher.NonceSize()))
+	ct := aeadCipher.Seal(nil, nonce, masterKey, []byte(passphrase))
+	ct = append(nonce, ct...)
+
+	return base58.Encode(ct), nil
+}
+
+func generateMasterKey(passphrase string) []byte {
+	// TODO consider using bcrypt: https://github.com/golang/crypto/tree/master/bcrypt
+	// mk, err := bcrypt.GenerateFromPassword([]byte(passphrase), bcrypt.MaxCost)
+	// if err != nil {
+	//  	return nil, err
+	// }
+	// instead of sha256 below - this means above logic must validate key length
+	masterKey := sha256.Sum256([]byte(passphrase))
```

Comment:
This should be some form of secret expansion - simply hashing the passphrase is generally insufficient.

Reply:
bcrypt in the comment above is an option, or we can use HKDF: https://godoc.org/golang.org/x/crypto/hkdf#ex-package--Usage

the aim, is this can be done in a separate PR

Reply:
done in master_secret.go .. we now use HKDF secret extraction